Userland/Services/SpiceAgent/Message.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Message.h"
+#include <AK/MemoryStream.h>
+#include <AK/Stream.h>
+#include <AK/String.h>
+
+namespace SpiceAgent {
+
+ErrorOr<void> Message::write_to_stream(AK::Stream&)
+{
+    return Error::from_string_literal("write_to_stream not implemented!");
+}
```

Comment:
Is it not fair to have a default in this case? I'm confused why this is a problem..? Is it not best for `Message` to have both and for the subclass to override whichever it actually uses?

Reply:
There is no real overriding happening here, these functions are not virtual. Subclasses can still define this function as they please, but I'd rather have it fail at compile time if we ever end up accidentally calling this function instead of one from one of the subclasses.